src/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -28,6 +28,11 @@ public static NotSupportedException GetNotSupportedException_SerializationNotSup
             return new NotSupportedException(SR.Format(SR.SerializationNotSupportedCollectionType, propertyType));
         }
 
+        public static void ThrowInvalidOperationException_SerializerCycleDetected(int maxDepth)
```

Comment:
The other methods have `[MethodImpl(MethodImplOptions.NoInlining)]`, but this one doesn't. Is that a mistake?

Reply:
The others really shouldn't have NoInlining if they're just doing a throw. It can actually make the codegen worse.

Reply:
Yes I avoided it here based on other recent feedback but didn't change the others.